### ISSUE TITLE
max-len does not properly ignore trailing comments

### ISSUE THREAD

**Issue by OliverSieweke** [eslint/eslint#11838]
**Tell us about your environment**

* **ESLint Version:** 5.16.0
* **Node Version:** 10.16.0 (current LTS)
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

Minimal working example for `.eslintrc.json`:
```js
{
    "rules": {
        "max-len": [
            "error",
            {
                "code": 20,
                "ignoreTrailingComments": true
            }
        ]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

`test.js`:
```js
"this is fine"; // Some comment... bla bla bla
"this is fine"; /* Some inline comment... */
"this isn't"; /* inline */ // Some other comment
```

```bash
eslint test.js
```

**What did you expect to happen?**
No error, as trailing comments should be ignored.

**What actually happened? Please include the actual, raw output from ESLint.**

```
  2:1  error  Line 2 exceeds the maximum line length of 20  max-len
```

**Comments**

This happens when the trailing comments include an inline as well as a line comment.

I'm happy to submit a pull request if this is recognized as a bug.

**Comment by mysticatea** [comment]
Thank you for your report.

Indeed, it sounds like a bug.
